assignments/introductory-assignments.md

```diff
@@ -488,14 +582,767 @@ A digital circuit computer with 2GB of memory can distinguish 2^{17 179 869 184}
+|	S11	|  = 27	|
+|	T3	|  = 28	|
+|	T4	|  = 29	|
+|	T5	|  = 30	|
+|	T6	|  = 31	|
+
+In the CPU, an [arithmetic logic unit](https://en.wikipedia.org/wiki/Arithmetic_logic_unit) (ALU) is a combinational digital circuit that performs arithmetic and [bitwise operations](https://en.wikipedia.org/wiki/Bitwise_operation) on integer binary numbers.
+The inputs to an ALU are the data to be operated on, called [operands](https://en.wikipedia.org/wiki/Operand), and a code indicating the operation to be performed; the ALU's output is the result of the performed operation. 
+
+#### Examples
+
+When we talk about the von Neumann architecture in Selfie, one would ask where it is defined or how we could read the different parameters up?
+
+In the Github Repository, we can look into the `selfie.c` file to clear that up. 
+
+For example, if we want to know how many bits the CPU architecture is defined, we could search for `cpubitwidth`, and there we can find `uint64_t CPUBITWIDTH = 64; // double word` that means the CPU's bit-width is 64 or 8-byte. The comment `// double word` will explain under the topic *Random access memory (RAM)*.
+
+#### Exercises
+
+Search in the `selfie.c` file: 
+
+- How many registers are implemented in Selfie? 
+- What is the size of a register in Selfie?
+
+### Program Counter (PC)
+
+How does a computer "know" what to execute? After all, the bits in memory could mean anything. They could be code; they could be data, anything. But the answer to that question can anyway not be any more straightforward.
+
+Processors based on the von Neumann architecture feature a special-purpose register as part of their control unit called the *program counter* (PC). The PC in Selfie is a special-purpose 64-bit register.
+
+[Program Counter (PC)](https://en.wikipedia.org/wiki/Program_counter): A processor register that indicates where a computer is in its program sequence. In most processors, the PC is incremented after fetching an instruction and holds the memory address of ("points to") the next instruction that would be executed. Instructions are usually fetched sequentially from memory, but control transfer instructions change the sequence by placing a new PC value. These include branches (sometimes called jumps), subroutine calls, and returns. 
+
+- A *control transfer* on some assertion's truth lets the computer follow a different sequence under different conditions. 
+- A *branch* provides that the next instruction is fetched from somewhere else in memory. 
+- A *subroutine call* not only branches but saves the preceding contents of the PC somewhere. 
+- A *return* retrieves the PC's saved contents and places it back in the PC, resuming sequential execution with the instruction following the subroutine call.
+
+Each instruction instructs the processor to perform some computation and determines the next value of the PC so that the processor "knows" where in memory the next instruction is stored. That sequence of PC values is called *control flow*.
```

Comment:
```suggestion
Each instruction instructs the processor to perform some computation and determines the next value of the PC so that the processor knows where in memory the next instruction is stored. That sequence of PC values is called control flow.
```